### PR TITLE
refine format of dynamic config doc

### PR DESCRIPTION
**Pull request by TomShawn**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: dynamic-config.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ This section describes the common operations of dynamic configuration change.
 
 To view the configuration of all instances in the cluster, use the `show config` SQL statement. The result is as follows:
 
-```
+```sql
 show config;
```
Comment (yikeke) [inline dynamic-config.md:28]:
```suggestion
show config;
```

```
```

Comment (yikeke) [inline dynamic-config.md:28]: Separate this code from the execution result.

File: dynamic-config.md [hunk 2]
```diff
@@ -95,114 +115,114 @@ This section lists the parameters supported by dynamic configuration change.
...
+| `pd-server.metric-storage` | Sets the storage address of the cluster metrics |
+| `pd-server.dashboard-address` | Sets the dashboard address |
+| `replication-mode.replication-mode` | Sets the backup mode |
 
 For detailed parameter description, refer to [PD Configuration File](/pd-configuration-file.md).
 
 ### TiKV
 
 | Parameter | Description |
 | --- | --- |
```
Comment (TomShawn) [inline dynamic-config.md:161]:
```suggestion
| :--- | :--- |
```

File: dynamic-config.md (hunk 1 above)
Comment (TomShawn) [reply dynamic-config.md:30]: Done. PTAL.

File: dynamic-config.md [hunk 3]
```diff
@@ -60,18 +65,35 @@ set config "127.0.0.1:2379" log.level="info"
 
 If the modification is successful, `Query OK` is returned:
 
-```
+{{< copyable "sql" >}}
+
+```sql
 set config '127.0.0.1:2379' log.level='info';
+```
```
Comment (yikeke) [inline dynamic-config.md:72]:
```suggestion
```

File: dynamic-config.md [hunk 4]
```diff
@@ -60,18 +65,30 @@ set config "127.0.0.1:2379" log.level="info"
 
 If the modification is successful, `Query OK` is returned:
 
-```
-set config '127.0.0.1:2379' log.level='info';
+
```
Comment (TomShawn) [inline dynamic-config.md:68]:
```suggestion
```

**Pull request merged**